.github/workflows/test-pull-request.yml

```diff
@@ -1,89 +1,122 @@
-######################################################
-on: pull_request
+    test:
+      runs-on: ubuntu-latest
+      needs: lint
+      permissions:
+        contents: read
+        statuses: write
+      steps:
+        - name: Checkout Code
+          uses: actions/checkout@v4
+  
+        - name: Setup Node
+          uses: actions/setup-node@v3
+          with:
+            node-version-file: '.nvmrc'
+            cache: 'yarn'
+            cache-dependency-path: 'yarn.lock'
+  
+        - name: Enable Corepack for Yarn
+          run: corepack enable
+  
+        - name: Install Dependencies
+          run: yarn install
+          env:
+            YARN_ENABLE_IMMUTABLE_INSTALLS: false
+  
+        - name: Check links
+          run: yarn test
 
-###############
-# Set the Job #
-###############
-jobs:
-  lint:
-    # Set the agent to run on
-    runs-on: ubuntu-latest
-
-    ############################################
-    # Grant status permission for MULTI_STATUS #
-    ############################################
-    permissions:
-      contents: read
-      packages: read
-      statuses: write
-
-    ##################
-    # Load all steps #
-    ##################
-    steps:
-      ##########################
-      # Checkout the code base #
-      ##########################
-      - name: Checkout Code
-        uses: actions/checkout@v4
-        with:
-          # Full git history is needed to get a proper list of changed files
-          # within `super-linter`
-          fetch-depth: 0
-      - name: Load super-linter configuration
-        run: cat .github/super-linter.env >> "$GITHUB_ENV"
-
-      ################################
-      # Run Linters against code base #
-      ################################
-      - name: Lint Code Base
-        #
-        # Use full version number to avoid cases when a next
-        # released version is buggy
-        # About slim image: https://github.com/github/super-linter#slim-image
-        uses: super-linter/super-linter/slim@v6.7.0
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          DEFAULT_BRANCH: main
-  test_and_build:
-    runs-on: ubuntu-latest
-    needs: lint
-    steps:
-      - name: Checkout Code
-        uses: actions/checkout@v4
-
-      - name: Setup Node
-        uses: actions/setup-node@v3
-        with:
-          node-version-file: '.nvmrc'
-          cache: 'yarn'
-          cache-dependency-path: 'yarn.lock'
-
-      - name: Enable Corepack for Yarn
-        run: corepack enable
-
-      - name: Install Dependencies
-        run: yarn install
-        env:
-          YARN_ENABLE_IMMUTABLE_INSTALLS: false
-
-      - name: Check links
-        run: yarn test
-
-      - name: Build site
-        run: yarn build
+    build:
+      runs-on: ubuntu-latest
+      needs: test
+      permissions:
+        contents: read
+        statuses: write
+      steps:
+        - name: Checkout Code
```

Comment:
Is it necessary to duplicate/repeat these steps? They've already been performed in the `test` job.

Reply:
Every job is run from scratch by design.

Reply:
For what purpose?

Reply:
By GitHub design.

Reply:
Found a way to avoid the duplication via a composite action.

Reply:
Nice work!